proxy_ring_buffer.go

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package hazelcast
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/hazelcast/hazelcast-go-client/internal/proto/codec"
+	iserialization "github.com/hazelcast/hazelcast-go-client/internal/serialization"
+)
+
+// A Ringbuffer is a data structure where the content is stored in a ring-like
+// structure. A Ringbuffer has a fixed capacity, so it won't grow beyond
+// that capacity and endanger the stability of the system. If that capacity
+// is exceeded, the oldest item in the Ringbuffer is overwritten.
+//
+// For details see https://docs.hazelcast.com/imdg/latest/data-structures/ringbuffer
+type Ringbuffer struct {
+	*proxy
+	partitionID int32
```

Comment:
It would be great to store the capacity of Ringbuffer as in Java client. 

https://github.com/hazelcast/hazelcast/blob/14ef0dbefa4a6fa00a50a8873aeea02ec6c3f5e4/hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientRingbufferProxy.java#L68

It makes a "lazy-set" for that field

Reply:
will do.

Reply:
Added as suggested.